ios/MVVM_Testing.md

```diff
@@ -0,0 +1,198 @@
+
+The `searchFor` function just returns the results from the NetworkingLayer, however in a more complex example, the SearchViewModel could perform sorting or filtering logic on the results. In that case the test would also include test coverage for such functionality.
+
+It is important to not test too much. The tests written against the view model should only test the actual functionality of the view model itself and not the model or other services accessed by the view model. For instance, in the above example, the test should not address the interworkings of the NetworkLayer. Instead, the test should focus on the view model's use of the results returned from the NetworkLayer. Other tests can be written directed at the NetworkingLayer that will not include the view model.
+
+The final method to test on the SearchViewModel is `albumViewModelAt`.
+
+```
+func albumViewModelAt(index: Int) -> AlbumSearchResultCellViewModel {
+	return albumViewModels[index]
+}
+```
+
+Testing the `albumViewModelAt` method is fairly straight forward and can be accomplished by simply creating a SearchViewModel with a known order of album elements and testing to ensure that the correct album is returned for each index, as shown below.
+
+```func testAlbumViewModelAtIndex() {
+
+        let album1 = Album(title: "Big Boat", artist: "Phish", releaseDate: Date(), imageURL: URL(string: "http://www.image1.jpg")!)
+        let album2 = Album(title: "Farmhouse", artist: "Phish", releaseDate: Date(), imageURL: URL(string: "http://www.image2.jpg")!)
+        let album3 = Album(title: "Billy Breathes", artist: "Phish", releaseDate: Date(), imageURL: URL(string: "http://www.image3.jpg")!)
+
+        let searchViewModel = SearchViewModel()
+        searchViewModel.albumViewModels = [AlbumSearchResultCellViewModel(album: album1), AlbumSearchResultCellViewModel(album: album2),AlbumSearchResultCellViewModel(album: album3)]
+
+        XCTAssertEqual(searchViewModel.albumViewModelAt(index: 0).title, "Big Boat")
+        XCTAssertEqual(searchViewModel.albumViewModelAt(index: 1).title, "Farmhouse")
+        XCTAssertEqual(searchViewModel.albumViewModelAt(index: 2).title, "Billy Breathes")
+    }
+```
+
+### Conclusion
+
+As shown in the above examples, testing view models can be a good first step to provide test coverage for presentation and other basic aspects of an apps business logic. When writing tests it is important to keep the following points in mind:
+
+- Use a view model’s public API as a guide for what tests should/could be written
+
+- Only test functionality of the view model, do not attempt to test the functionality of any dependencies of the view model, use dependency injection if needed
+
+The entire app codebase for this test app can be found [here](https://github.com/weberbry/ViewModelTestingDemo).
```

Comment:
I've created a repo for MVVM examples:

https://github.com/IntrepidPursuits/MVVMExamples-ios

Lets consolidate these?